pac4cli/wpad.py

```diff
@@ -129,27 +130,57 @@ def get_config_wpad_url(self, config_file):
             logger.info("No wpad url specified")
             return None
 
+    @inlineCallbacks
+    def get_dns_wpad_urls(self, domains):
+        dns_urls = []
+        for domain in domains:
+            # We need to be aware of the TLD in the passed domains, so as to 
+            # avoid looking for the wpad url outside the business/company/entity
+            # e.g.: for http://hostname.subdomain.example.co.uk, we want to try: 
+            # http://wpad.hostname.subdomain.example.co.uk/wpad.dat
+            # http://wpad.subdomain.example.co.uk/wpad.dat
+            # http://wpad.example.co.uk/wpad.dat
+            # and avoid trying:
+            # http://wpad.co.uk/wpad.dat
+            #
+            # The tld package is using Mozilla's database for top level domains.
+            logger.info("Found fqdn: '%s'", domain)
+            tld_res = yield get_tld(domain, as_object=True, fix_protocol=True)
```

Comment:
get_tld is not an asynchronous function, is it? `yield` probably still works but I'm not 100% sure it does.

Reply:
You are right. twisted still confuses me :D